examples/inverse/plot_morph_VolSourceEstimate.py

```diff
@@ -0,0 +1,282 @@
+
+    # Pick the channels of interest
+    raw.pick_channels([raw.ch_names[pick] for pick in picks])
+
+    # Re-normalize our empty-room projectors, so they are fine after
+    # subselection
+    raw.info.normalize_proj()
+    # Read epochs
+    proj = False  # already applied
+    epochs = mne.Epochs(raw, events, event_id, tmin, tmax,
+                        baseline=(None, 0), preload=True, proj=proj,
+                        reject=dict(grad=4000e-13, mag=4e-12, eog=150e-6))
+    evoked = epochs.average()
+
+    # Read regularized noise covariance and compute regularized data covariance
+    noise_cov = mne.compute_covariance(epochs, tmin=tmin, tmax=0,
+                                       method='shrunk')
+    data_cov = mne.compute_covariance(epochs, tmin=0.04, tmax=0.15,
+                                      method='shrunk')
+    # Read forward model
+    forward = mne.read_forward_solution(fname_fwd)
+
+    # Compute weights of free orientation (vector) beamformer with weight
+    # normalization (neural activity index, NAI). Providing a noise covariance
+    # matrix enables whitening of the data and forward solution. Source
+    # orientation is optimized by setting pick_ori to 'max-power'.
+    # weight_norm can also be set to 'unit-noise-gain'. Source orientation can
+    # also be 'normal' (but only when using a surface-based source space) or
+    # None, which computes a vector beamfomer. Note, however, that not all
+    # combinations of orientation selection and weight normalization are
+    # implemented yet.
+    filters = make_lcmv(evoked.info, forward, data_cov, reg=0.05,
+                        noise_cov=noise_cov, pick_ori='max-power',
+                        weight_norm='nai')
+
+    # Apply this spatial filter to the evoked data.
+    stc = apply_lcmv(evoked, filters, max_ori_out='signed')
+
+    # take absolute values for plotting
+    stc.data[:, :] = np.abs(stc.data)
+
+    # Save result in stc files if desired
+    if fname is not None:
+        stc.save('lcmv-vol')
+
+    # select time window (tmin, tmax) in ms - consider changing for real data
+    # scenario, since those values were chosen to optimize computation time
+    stc.crop(0.087, 0.087)
+
+    src = forward['src']
+
+    # Save result in a 4D nifti file
+    img = mne.save_stc_as_volume(fname, stc, src,
+                                 mri_resolution=True)
+
+    return img, stc, src
+
+
+###############################################################################
+# Save non linear mapping data
+
+def write_morph(fname, morph, overwrite=True):
+    morph_out = dict()
+
+    # dissolve object structure
+    for key, value in morph.items():
+        # save type for order independent decomposition
+        if hasattr(value, '__dict__'):
+            value = value.__dict__
+        morph_out[key] = value
+    if not fname.endswith('.h5'):
+        fname += '.h5'
+    write_hdf5(fname, morph_out, overwrite=overwrite)
```

Comment:
I think I said this elsewhere -- you know what `morph` should contain so populate it directly. It's much more explicit

Reply:
I cannot hand over the ``dipy`` objects directly (rejected by ``write_hdf5``) so I "have to" extract the basic objects and reassemble after loading... Which would be a better way?